### PR TITLE
Added MusicBrainz Recording Id + Scrobbling

### PR DESCRIPTION
**Pull request by smoralis**
_Please provide description for this PR_
The PR adds the ability to read Musicbrainz Recording ID from the files as to scrobble it with Simple Last Fm.

_what it does_
It reads the metadata from mp3 (either the UFID value ot the TXXX value if defined) / vorbis comment MUSICBRAINZ_TRACKID / mpeg4 value Musicbrainz Track Id and stores it in the db in the field MBID.
Called with Song.MBID 

_why was it introduced_ 
https://community.metabrainz.org/t/android-player-mbid-submission-support/446353/6
https://github.com/simple-last-fm-scrobbler/sls/issues/421

_how to use it._
Files must be tagged with the Musicbrainz Identifier , and simple-last-fm-scrobbler must be installed for the PR to scrobble it.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,7 +1,6 @@
-#Wed Jun 27 14:32:00 EDT 2018
+#Wed Oct 30 21:16:00 EET 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-4.8.1-all.zip
-distributionSha256Sum=ce1645ff129d11aad62dab70d63426fdce6cfd646fa309dc5dc5255dd03c7c11
```
Comment (Adonai) [inline gradle/wrapper/gradle-wrapper.properties]: Not needed

File: app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java [hunk 2]
```diff
@@ -1174,18 +1174,19 @@ private void scrobbleBroadcast() {
...
 			sendBroadcast(intent);
 		}
 
 		if (mScrobble) {
 			Intent intent = new Intent("net.jjc1138.android.scrobbler.action.MUSIC_STATUS");
 			intent.putExtra("playing", (mState & FLAG_PLAYING) != 0);
-			if (androidIds[0] != -1)
-				intent.putExtra("id", androidIds[0]);
+			intent.putExtra("secs", (int)song.duration / 1000);
+			intent.putExtra("clashWithScrobbleDroid", "1");
```
Comment (Adonai) [inline app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java]: What does this line mean?

File: app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java [hunk 3]
```diff
@@ -1174,18 +1174,19 @@ private void scrobbleBroadcast() {
 			intent.putExtra("track", song.title);
 			intent.putExtra("album", song.album);
 			intent.putExtra("artist", song.artist);
-			if (androidIds[0] != -1) {
-				intent.putExtra("songid", androidIds[0]);
```
Comment (Adonai) [inline app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java]: Why were those deleted?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (smoralis) [reply gradle/wrapper/gradle-wrapper.properties]: Got this warning in Android Studio 

WARNING: It is not fully supported to define distributionSha256Sum in gradle-wrapper.properties.
Using an incorrect value may freeze or crash Android Studio.
Please manually verify or remove this property from all of included projects if applicable.
For more details, see https://github.com/gradle/gradle/issues/9361.

Use "ce1645..." as checksum for https://services.gradle.org/distributions/gradle-4.8.1-all.zip and sync project
Remove distributionSha256Sum and sync project
Open Gradle wrapper properties

File: app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java (hunk 2 above)
Comment (smoralis) [reply app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java]: https://github.com/simple-last-fm-scrobbler/sls/blob/294ecfef30b386e5e3d6cffc39a987f2bcdfe7fb/app/src/main/java/com/adam/aslfms/receiver/MusicAPI.java#L119

File: app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java (hunk 3 above)
Comment (smoralis) [reply app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java]: My bad , no need to remove for scrobbleBroadcast()

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (adrian-bl) [reply gradle/wrapper/gradle-wrapper.properties]: I really don't care about Android Studio bugs: Please keep the checksum around

File: app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java (hunk 2 above)
Comment (adrian-bl) [reply app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java]: Can you elaborate? I don't understand the functionality.

Comment (smoralis) [reply app/src/main/java/ch/blinkenlights/android/vanilla/PlaybackService.java]: It's just a boolean that the app can also use the unmaintained Scrobble Droid app https://github.com/JJC1138/scrobbledroid
It can be removed if requested, I guess most are using simple last fm scrobbler now.

**Comment by MrClon** [comment]
Will it be merged? I need this feature

**Comment by MrClon** [comment]
Will it be merged? I need this feature

**Comment by MrClon** [comment]
Will it be merged? I need this feature

**Comment by MrClon** [comment]
Will it be merged? I need this feature

**Comment by MrClon** [comment]
Will it be merged? I need this feature

**Comment by MrClon** [comment]
Will it be merged? I need this feature

**Comment by MrClon** [comment]
Will it be merged? I need this feature

**Comment by Adonai** [comment]
No idea, I'm not the maintainer

**Comment by petitminion** [comment]
any news on this ? This feature is great

**Review by adrian-bl — COMMENTED** [review]
> any news on this ? This feature is great

No: none of my comments were adressed.

File: app/src/main/java/ch/blinkenlights/android/medialibrary/MediaSchema.java [hunk 4]
```diff
@@ -36,7 +36,8 @@
 	  + MediaLibrary.SongColumns.MTIME        +" TIMESTAMP DEFAULT (strftime('%s', CURRENT_TIMESTAMP)), "
 	  + MediaLibrary.SongColumns.DURATION     +" INTEGER NOT NULL, "
 	  + MediaLibrary.SongColumns.PATH         +" VARCHAR(4096) NOT NULL, "
-	  + MediaLibrary.SongColumns.FLAGS        +" INTEGER NOT NULL DEFAULT 0 "
+	  + MediaLibrary.SongColumns.FLAGS        +" INTEGER NOT NULL DEFAULT 0, "
+	  + MediaLibrary.SongColumns.MBID         +" TEXT"
```
[the remark this answers is not in the document]
Comment (petitminion) [reply app/src/main/java/ch/blinkenlights/android/medialibrary/MediaSchema.java:40]: Mbid are stings with lettre ans integers so I suppose this need to stay as a text entire.
https://musicbrainz.org/doc/MusicBrainz_Identifier

File: app/src/main/java/ch/blinkenlights/android/medialibrary/MediaMetadataExtractor.java [hunk 5]
```diff
@@ -371,11 +372,11 @@ private void extractMetadata(String path) {
 	 */
 	private void populateSelf(HashMap bastp) {
 		// mapping between vorbiscomment -> constant
-		String[] map = new String[]{ "TITLE", TITLE, "ARTIST", ARTIST, "ALBUM", ALBUM, "ALBUMARTIST", ALBUMARTIST, "COMPOSER", COMPOSER, "GENRE", GENRE,
+		String[] map = new String[]{ "TITLE", TITLE, "ARTIST", ARTIST, "ALBUM", ALBUM, "ALBUMARTIST", ALBUMARTIST, "COMPOSER", COMPOSER, "GENRE", GENRE,"MUSICBRAINZ_TRACKID" , MBID,"MUSICBRAINZ TRACK ID" , MBID,
 		                             "TRACKNUMBER", TRACK_NUMBER, "TRACKTOTAL", TRACK_COUNT, "DISCNUMBER", DISC_NUMBER, "DISCTOTAL", DISC_COUNT,
 		                             "YEAR", YEAR };
 		// switch to integer filter if i >= x
-		int filterByIntAt = 12;
+		int filterByIntAt = 15;
```
[the remark this answers is not in the document]
Comment (petitminion) [reply app/src/main/java/ch/blinkenlights/android/medialibrary/MediaMetadataExtractor.java:379]: I count 15, there is only 3 new entries 12+3=15

File: app/src/main/java/ch/blinkenlights/android/medialibrary/MediaLibrary.java [hunk 6]
```diff
@@ -653,6 +653,7 @@ public static long hash63(String str) {
 		 * Various flags of this entry, see SONG_FLAG...
 		 */
 		String FLAGS = "_flags";
+		String MBID = "mbid";
```
[the remark this answers is not in the document]
Comment (petitminion) [reply app/src/main/java/ch/blinkenlights/android/medialibrary/MediaLibrary.java:656]: Can't edit but : 
/**
* Musicbrainz I'd of the song. Used by the scrobblers
*/

File: app/src/main/java/ch/blinkenlights/android/medialibrary/MediaSchema.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (chpio) [reply app/src/main/java/ch/blinkenlights/android/medialibrary/MediaSchema.java:40]: MbIds are Uuids, they can theoretically be represented as a 128 bit int.

**Comment by Freso** [comment]
Hi @smoralis, why did you close this? If you’re not interested in continuing work on this, I might take a stab at reviving this.

**Comment by smoralis** [comment]
Hi, I had no response.
Go ahead with it, if you want.

**Comment by petitminion** [comment]
> Hi @smoralis, why did you close this? If you’re not interested in continuing work on this, I might take a stab at reviving this.

I think you can, I tried to reach out already here and at https://community.metabrainz.org/t/android-player-mbid-submission-support/446353

I you understand why it do not work of would be greeeat ! <3
